src/System.IO.Compression.ZipFile/src/System/IO/Compression/ZipFileExtensions.cs

```diff
@@ -0,0 +1,296 @@
+        /// <exception cref="ArgumentNullException">destinationDirectoryName is null.</exception>
+        /// <exception cref="PathTooLongException">The specified path, file name, or both exceed the system-defined maximum length.
+        /// For example, on Windows-based platforms, paths must be less than 248 characters, and file names must be less than 260 characters.</exception>
+        /// <exception cref="DirectoryNotFoundException">The specified path is invalid, (for example, it is on an unmapped drive).</exception>
+        /// <exception cref="IOException">An archive entry’s name is zero-length, contains only white space, or contains one or more invalid
+        /// characters as defined by InvalidPathChars. -or- Extracting an archive entry would have resulted in a destination
+        /// file that is outside destinationDirectoryName (for example, if the entry name contains parent directory accessors).
+        /// -or- An archive entry has the same name as an already extracted entry from the same archive.</exception>
+        /// <exception cref="UnauthorizedAccessException">The caller does not have the required permission.</exception>
+        /// <exception cref="NotSupportedException">destinationDirectoryName is in an invalid format. </exception>
+        /// <exception cref="InvalidDataException">An archive entry was not found or was corrupt.
+        /// -or- An archive entry has been compressed using a compression method that is not supported.</exception>
+        /// 
+        /// <param name="destinationDirectoryName">The path to the directory on the file system.
+        /// The directory specified must not exist. The path is permitted to specify relative or absolute path information.
+        /// Relative path information is interpreted as relative to the current working directory.</param>
+        public static void ExtractToDirectory(this ZipArchive source, String destinationDirectoryName)
+        {
+            if (source == null)
+                throw new ArgumentNullException("source");
+
+            if (destinationDirectoryName == null)
+                throw new ArgumentNullException("destinationDirectoryName");
+
+            Contract.EndContractBlock();
+
+            // Rely on Directory.CreateDirectory for validation of destinationDirectoryName.
+
+            // Note that this will give us a good DirectoryInfo even if destinationDirectoryName exists:
+            DirectoryInfo di = Directory.CreateDirectory(destinationDirectoryName);
+            String destinationDirectoryFullPath = di.FullName;
+
+            foreach (ZipArchiveEntry entry in source.Entries)
+            {
+                String fileDestinationPath = Path.GetFullPath(Path.Combine(destinationDirectoryFullPath, entry.FullName));
+
+                if (!fileDestinationPath.StartsWith(destinationDirectoryFullPath, StringComparison.OrdinalIgnoreCase))
```

Comment:
Thanks Stephen I don't think so.

My first thought was to move GetComparison to corefx\src\Common\src\System\IO\PathInternal.Unix.cs (and PathInternal.Windows.cs and add a new PathInternal.OSX.cs) and remove FileSystem.CaseSensitive.

But I from issue 1086 I see you might want this class in the future to decide based on what kind of file system it wraps rather than what platform it's on.

It seems like we may want to consider a public API in the future if other libraries are going to need this information.  Until then, though, how do you feel about me moving this to the common PathInternal?